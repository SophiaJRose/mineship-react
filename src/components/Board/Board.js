import * as React from 'react'
import { Tile } from '../Tile'

class Board extends React.Component {
	render() {
		return (
			<table>
				<tbody>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
					<tr>
						<td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td><td><Tile /></td>
					</tr>
				</tbody>
			</table>
		);
	}
}

export { Board }