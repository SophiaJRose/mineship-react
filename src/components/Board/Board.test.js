import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { Board } from "./Board"

describe("<Board /> tests", () => {
	it("should render as 8x8 grid of tiles with background", () => {
		const { container } = render(<Board />)
		expect(container.firstChild).toMatchSnapshot()
	})
})