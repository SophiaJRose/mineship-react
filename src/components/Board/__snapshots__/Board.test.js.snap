// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> tests should render as 8x8 table of tiles 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
    <tr>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
      <td>
        <button />
      </td>
    </tr>
  </tbody>
</table>
`;
